Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -424,31 +424,57 @@ public final class OAuth2Client {
         
         // Exchange the token
         request.send(to: self) { result in
-            guard case let .success(response) = result else {
-                completion(result)
-                return
-            }
-            
-            // Perform idToken/accessToken validation
-            do {
-                try response.result.validate(using: self, with: request as? IDTokenValidatorContext)
-            } catch {
-                completion(.failure(.validation(error: error)))
-                return
-            }
-            
             // Wait for the JWKS keys, if necessary
-            group.notify(queue: DispatchQueue.global()) {
+            group.notify(queue: DispatchQueue.global()) { [weak self] in
+                // Perform idToken/accessToken validation
+                self?.validateToken(
+                    request: request,
+                    keySet: keySet,
+                    oauthTokenResponse: result,
+                    completion: completion
+                )
+            }
```

Comment:
Can you put the first argument inline, so the other methods indent along with them?
```suggestion
                self?.validateToken(request: request,
                                    keySet: keySet,
                                    oauthTokenResponse: result,
                                    completion: completion)
            }
```
```suggestion
                self?.validateToken(
                    request: request,
                    keySet: keySet,
                    oauthTokenResponse: result,
                    completion: completion
                )
            }
```

Reply:
Sounds good.